### ISSUE TITLE
split features

### ISSUE THREAD

**Issue by renanbr** [bjori/mongo-php-transistor#9]
I've made some experiments with @alfsb, then this idea come from these experiences.
Features could be splitted on traits, so users could pick which attach to their classes.

A full-featured persistable class sample:

``` php
class Document implements MongoDB\BSON\Persistable
{
    use Persistent; // base trait, it provides persistable methods implementation
    use CreatedField; // provides $_created and getCreated()
    use LastModifiedField; // provides $_lastModified and getLastModified()
    use AutoGeneratedId; // provides $_id and getId()
    // and others, such as atomic update

    /* ... */
}
```

**Comment by bjori** [comment]
I did originally think about something like that, but it seemed to be overly complicated to check if this or that functionality was enabled when serializing. And when adding new traits into the mix, the base trait would always have to be specifically told about it.

If you can come up with a way that seamlessly can support it, then I think that would be pretty cool!

**Comment by renanbr** [comment]
I've made an initial work at the [split branch](https://github.com/renanbr/mongo-php-transistor/tree/split) to try the concept out.

Important notes:
- `atomic update` isn't yet implemented, I'm waiting for [PHPC-550](https://jira.mongodb.org/browse/PHPC-550) consequences
- [Test fail](https://travis-ci.org/renanbr/mongo-php-transistor/builds/123500805)
  - 3, 4 and 7 fail because `atomic update` isn't available, so `replace update` gives always a value to the __lastModified field
  - and 6 fails because _id receive a value just when getId() or bsonSerialize()

**Comment by bjori** [comment]
I think those tests already fail in the current branch, don't they?
@jmikola told me they were going to remove support for atomic updates

**Comment by jmikola** [comment]
Correct. There are no plans to change the persistence spec or driver here. The ODS field will always injected into the `bsonSerialize()` return value if the class implements `MongoDB\BSON\Persistable`.

@derickr was actually just talking about removing the two ODS tests that we marked as `XFAIL`.
